Mathlib/Algebra/Ring/Hom/Defs.lean

```diff
@@ -611,6 +611,14 @@ theorem cancel_left {g : β →+* γ} {f₁ f₂ : α →+* β} (hg : Injective
     g.comp f₁ = g.comp f₂ ↔ f₁ = f₂ :=
   ⟨fun h => RingHom.ext fun x => hg <| by rw [← comp_apply, h, comp_apply], fun h => h ▸ rfl⟩
 
+lemma precomp_injective_of_surjective {α β γ : Type*} [NonAssocSemiring α]
+    [NonAssocSemiring β] [NonAssocSemiring γ] (f : α →+* β)
+    (hf : Function.Surjective f) {g₁ g₂ : β →+* γ} (h : g₁.comp f = g₂.comp f) :
+    g₁ = g₂ := by
+  ext s
+  obtain ⟨r, rfl⟩ := hf s
+  simp only [← comp_apply, h]
```

Comment:
```suggestion
    [NonAssocSemiring β] [NonAssocSemiring γ] (f : α →+* β) (hf : Function.Surjective f) :
    Function.Injective fun g : β →+* γ ↦ g.comp f :=
  fun _ _ ↦ (cancel_right hf).mp
```
Maybe?

Reply:
It would be good to find the `Function` version of this lemma, and match the naming.

Reply:
That lemma is `Function.Surjective.injective_comp_right`, so I guess this could be `Function.Surjective.injective_ringComp_right` or similar?

Reply:
You can even prove this one in terms of it, as
```lean
.of_comp (f := DFunLike.coe) <| hf.injective_comp_right.comp DFunLike.coe_injective
```

Reply:
Thanks! I do not know how I did not see `cancel_right`, which I am not using instead.